### PR TITLE
Postorder traversal added

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#179 )

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Please review sir @HarshCasper @atarax665 @ankitaggarwal23 @gat786

**Comment by shivammahajancse** [comment]
Sir @iamrajiv please review

**Comment by shivammahajancse** [comment]
Please review @nidh-ish

**Comment by nidh-ish** [comment]
@shivammahajancse Make similar changes here as you were asked to do in the in-order and post-order traversal. Run a Grammarly check also.

**Comment by shivammahajancse** [comment]
All changes done @nidh-ish

File: docs/data-structures/postordertraversal.mdx [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+title: Postorder traversal of Binary Tree
+slug: /data-structures/postordertraversal
+---
+
+## Introduction
+
+Binary Tree is a type of tree which have at most two children or we can say at most degree 2. So to traverse the binary tree according to Depth-first search. We have a traversal algorithm called Postorder traversal.
```
Comment (nidh-ish) [inline docs/data-structures/postordertraversal.mdx:8]: Make it: A Binary Tree

File: docs/data-structures/postordertraversal.mdx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+## Explanation 
+In this type of traversal(Postorder traversal). first, we have to visit left part then right part then its root respectively. In this first visit left part that means we have to traverse the whole left subtree in this order (left->right->root) after that whole right subtree in this order (left->right->root) and then visit root of that subtree. Postorder: (left->right->root).
```
Comment (nidh-ish) [inline docs/data-structures/postordertraversal.mdx:11]: Make it:

In this type of traversal(Postorder traversal). first, we have to visit the left part then the right part then its root respectively. In this first visit left part that means we have to traverse the whole left subtree in this order (left->right->root) after that whole right subtree in this order (left->right->root) and then visit the root of that subtree. Postorder: (left->right->root).

File: docs/data-structures/postordertraversal.mdx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+For Postorder traversal, we follow the following algorithm:
+
+* First, we have to traverse the Binary tree in this order(left->right->root).
+
+* Left means traversing the whole left subtree in this order(left->right->root) until we get left child as null. 
+
+* Root means parent of that particular subtree.
+
+* Right means traverse the whole right subtree in this (left->right->root) order until we get null.
```
Comment (nidh-ish) [inline docs/data-structures/postordertraversal.mdx:47]: Make it: For Postorder traversal, we follow the following algorithm:

First, we have to traverse the Binary tree in this order(left->right->root).

Left means traversing the whole left subtree in this order(left->right->root) until we get the left child as null.

Root means the parent of that particular subtree.

Right means traverse the whole right subtree in this (left->right->root) order until we get null.

File: docs/data-structures/postordertraversal.mdx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+#### Input :
+
+```
+Enter the data: 44 22 66 11 33 55 77 
+
+```
+#### Output :
+
+```
+Postorder traversal : 11 33 22 55 77 66 44
```
Comment (nidh-ish) [inline docs/data-structures/postordertraversal.mdx:199]: Make stepwise sample input as it asks in the code.

**Review by nidh-ish — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Sir done @iamrajiv

**Review by iamrajiv — APPROVED** [review]

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Rename this file to `postorder_traversal.mdx`.

**Comment by shivammahajancse** [comment]
Sir done @iamrajiv

**Review by iamrajiv — APPROVED** [review]

File: sidebars.js [hunk 5]
```diff
@@ -48,7 +48,7 @@ module.exports = {
                 "data-structures/binary-tree",
                 "data-structures/Singly-Linked-list",
                  "data-structures/binary-search-tree",
-                "data-structures/preorder_traversal"
```
Comment (atarax665) [inline sidebars.js:51]: Do not delete ```data-structures/preorder_traversal```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Done sir @atarax665

**Review by iamrajiv — APPROVED** [review]

File: docs/data-structures/postorder_traversal.mdx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+ Postorder traversal: 11 33 22 55 77 66
+ Root of the right subtree: Node(44)
+ Now our Postorder traversal: 11 33 22 55 77 66 44 !!
+
+```
+## Algorithm
+
+For Postorder traversal, we follow the following algorithm:
+
+* First, we have to traverse the Binary tree in this order(left->right->root).
```
Comment (atarax665) [inline docs/data-structures/postorder_traversal.mdx:41]:
```suggestion
* First, we have to traverse the Binary tree in the order ```left -> right -> root```.
```

File: docs/data-structures/postorder_traversal.mdx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+* Left means traversing the whole left subtree in this order(left->right->root) until we get left child as null.
```
Comment (atarax665) [inline docs/data-structures/postorder_traversal.mdx:43]:
```suggestion
* Left means traversing the whole left subtree in this order ```left -> right -> root``` until we get a left child as null. 
```

File: docs/data-structures/postorder_traversal.mdx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+For Postorder traversal, we follow the following algorithm:
+
+* First, we have to traverse the Binary tree in this order(left->right->root).
+
+* Left means traversing the whole left subtree in this order(left->right->root) until we get left child as null. 
+
+* Root means parent of that particular subtree.
+
+* Right means traverse the whole right subtree in this (left->right->root) order until we get null.
```
Comment (atarax665) [inline docs/data-structures/postorder_traversal.mdx:47]:
```suggestion
* Right means traverse the whole right subtree in this ```left -> right -> root``` order until we get null.
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Sir all changes done @atarax665

File: docs/data-structures/postorder_traversal.mdx [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+For Postorder traversal, we follow the following algorithm:
+
+* First, we have to traverse the Binary tree in the order ```left -> right -> root```.
+
+* Left means traversing the whole left subtree in this order ```left -> right -> root``` until we get a left child as null. 
+
+* Root means parent of that particular subtree.
+
+* Right means traverse the whole right subtree in this <span class="x x-first x-last">```</span>left<span class="x x-first x-last"> -&gt; </span>right<span class="x x-first x-last"> -&gt; </span>root<span class="x x-first x-last">```</span> order until we get null.
```
Comment (atarax665) [inline docs/data-structures/postorder_traversal.mdx:50]: Remove all these tags.

Comment (shivammahajancse) [reply docs/data-structures/postorder_traversal.mdx]: Sir done @atarax665

**Comment by shivammahajancse** [comment]
@atarax665  sir please review

**Review by siddharth25pandey — APPROVED** [review]
@shivammahajancse LGTM 💯 Able to merge

**Pull request merged**